moto/iam/access_control.py

```diff
@@ -425,6 +475,15 @@ def is_action_permitted(
         if is_action_concerned:
             if self.is_unknown_principal(self._statement.get("Principal")):
                 return PermissionResult.NEUTRAL
+            elif principal and not self._check_principal(principal):
+                return PermissionResult.DENIED
+
+            if conditions and not self._check_conditions(conditions):
+                return PermissionResult.DENIED
+
+            if not self._statement.get("Resource"):
```

Comment:
This is only true for AssumeRolePolicies, right? For other policies, the Resource is always mandatory. Would be good to have some documentation (i.e, a simple comment) to explain why this if-statement holds true.

Reply:
Added comment before check